geomstats/learning/intrinsic_dimension_estimation.py

```diff
@@ -0,0 +1,100 @@
+"""Intrinsic Dimension Estimation."""
+
+import geomstats.backend as gs
+
+
+class LevinaBickelEstimator:
+    r"""Levina Bickel Estimator for intrinsic dimension estimation.
+
+    Levian Bickel Estimator calculates intrinsic dimension of
+    manifolds embedded in euclidean space through maximum likelihood
+    estimation.
+
+    .. math::
+        T_j(x_i)=\text{j th Nearest Neighbor to }  x_i \text{ in the data}
+
+        m_k(x_i)=[\frac{1}{k-1} \sum_{j=1}^{k}
+        \frac{\log T_k(x)}{\log T_{j}(x)}]^{-1}
+
+        m_k=\frac{1}{N} \sum_{i=1}^{N} m_{k}(x_i)
+
+        m=\frac{1}{k_2 - k_1 + 1} \sum_{k=k_1}^{k_2} m_k
+
+    Parameters
+    ----------
+    min_neighbors : int
+        Minimum Neighbors used for computing estimator
+        Optional, default: 5.
+    max_neighbors : int
+        Maximum Neighbors used for computing estimator
+        Optional, default: 20.
+
+    Example
+    -------
+    >>> from sklearn import  datasets
+    >>> import geomstats.backend as gs
+    >>> import geomstats.learning.intrinsic_dimension_estimation as ide
+    >>> from ide import LevinaBickelEstimator
+    >>> X, _ = datasets.make_swiss_roll(n_samples=1500)
+    >>> X = gs.array(X)
+    >>> lbe = LevinaBickelEstimator()
+    >>> predicted_intrinsic_dim =lbe.fit(X).predict()
+
+    Reference
+    ----------
+    .. [2004] . "Maximum Likelihood Estimation of Intrinsic Dimension"
+    Advances in Neural Information Processing Systems 17 (NIPS 2004)
+    https://arxiv.org/abs/1908.09326
+    """
+
+    def __init__(self, min_neighbors=5, max_neighbors=20):
+        self.min_neighbors = min_neighbors
+        self.max_neighbors = max_neighbors
+        self.log_sorted_dist = None
+
+    def fit(self, X):
+        """Calculate distance matrix
+
+        Parameters
+        ----------
+        X : array-like
```

Comment:
What is the shape here, probably [n_samples, dim]? Do we want to make it work for shapes such as [n_samples, n, m], i.e. for the manifolds of matrices or discrete landmarks/curves that we have?